pkg/loop/internal/core/services/capability/capabilities_registry.go

```diff
@@ -526,6 +557,29 @@ func (c *capabilitiesRegistryServer) Add(ctx context.Context, request *pb.AddReq
 	return &emptypb.Empty{}, nil
 }
 
+func (c *capabilitiesRegistryServer) Remove(ctx context.Context, request *pb.RemoveRequest) (*emptypb.Empty, error) {
+	conn, err := c.Dial(request.CapabilityID)
```

Comment:
Ah I understand now: I think this is a case where I wouldn't be afraid to just have Remove take an ID rather than the actual capability

I think that will simplify a lot, in the client you can just call Info() then Remove(info.ID) and propagate that through here -- no new clients, connections, etc. needed, since the call just operates on data.